File: core/src/main/java/com/a65apps/core/interactors/locations/PersonLocationRepository.java
package com.a65apps.core.interactors.locations;

import com.a65apps.core.entities.Location;

import java.util.List;

import io.reactivex.rxjava3.core.Single;

public interface PersonLocationRepository {

    Single<Location> getLocationByPerson(String personId);

    Single<List<Location>> getAllPersonLocation();

    Single <Location> createPersonLocation(Location location);

}

File: core/src/main/java/com/a65apps/core/interactors/locations/PersonLocationRepository.kt
package com.a65apps.core.interactors.locations

import com.a65apps.core.entities.Location
import kotlinx.coroutines.flow.Flow

interface PersonLocationRepository {
    fun getLocationByPerson(personId: String): Flow<Location?>
    fun getAllPersonLocation(): Flow<List<Location>>?
    fun createPersonLocation(location: Location)
}
